app/utils/MessageRenderer.scala

```diff
@@ -22,7 +22,7 @@ import play.api.i18n.Messages
 object MessageRenderer {
 
   def getMessageKey(msgKey: String)(implicit user: UserRequest[_]): String = {
-    if (user.isAgent && didClientCauseLateSubmission) {
+    if (user.isAgent && (didClientCauseLateSubmission || (user.session.get(SessionKeys.whoPlannedToSubmitVATReturn).isEmpty && user.session.get(SessionKeys.causeOfLateSubmissionAgent).isEmpty))) {
```

Comment:
msgKey evaluates to honestyDeclaration.lpp.li.1.2.myClient [when](url) user is agent  and the rest clause is false. I am guessing this would be handled in the PRM-498 to show correct msg when the 'Who Planned Submit Return' and 'Why was return submitted Late' removed.

![image](https://user-images.githubusercontent.com/86231805/128359172-06687772-2444-4643-8d35-0b5f30185136.png)

Reply:
Interesting how you got this message? It should be impossible for you to get `honestyDeclaration.lpp.li.1.2.myClient`, because this is an agent specific message, in the messages file. It would be interesting to see how you got this? Did you go through the 'who planned to submit the return' page and the 'why was return submitted late'?